frontend/src/components/login/login-callback/LoginCallback.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useContext, useEffect, useState } from 'react'
+
+import { AuthContext } from 'contexts/auth.context'
+
+import { getUser, loginWithSgid, setUserAnalytics } from 'services/auth.service'
+
+import { sendException } from 'services/ga.service'
+
+const DASHBOARD_PAGE = '/'
+const LOGIN_PAGE = '/ogp-login'
+
+const LoginCallback = () => {
+  const {
+    setAuthenticated,
+    setEmail: setAuthContextEmail,
+    setExperimentalData,
+  } = useContext(AuthContext)
+
+  const [isLoading, setIsLoading] = useState(false)
+  const params = new URL(window.location.href).searchParams
+  const code = params.get('code')
+
+  useEffect(() => {
+    async function login(code: string) {
+      setIsLoading(true)
+      try {
+        await loginWithSgid(code)
+        setAuthenticated(true)
```

Comment:
Why do we `setAuthenticated(true)` regardless of the result from `await loginWithSgid(code)`?

Reply:
`await loginWithSgid(code)` will throw an error if the request was unsuccessful. Therefore, `setAuthenticated(true)` will only run if the login was successful. Otherwise, we would execute the `catch` path.

Reply:
Ah, `errorHandler` in the catch clause rethrows error. Got it!